matlab/src/matlab/+arrow/+io/+ipc/RecordBatchStreamReader.m

```diff
@@ -0,0 +1,83 @@
+%RECORDBATCHSTREAMREADER Class for reading Arrow record batches from the
+% Arrow IPC Stream format.
+
+% Licensed to the Apache Software Foundation (ASF) under one or more
+% contributor license agreements.  See the NOTICE file distributed with
+% this work for additional information regarding copyright ownership.
+% The ASF licenses this file to you under the Apache License, Version
+% 2.0 (the "License"); you may not use this file except in compliance
+% with the License.  You may obtain a copy of the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+
+classdef RecordBatchStreamReader < matlab.mixin.Scalar
+
+    properties(SetAccess=private, GetAccess=public, Hidden)
+        Proxy
+    end
+
+    properties (Dependent, SetAccess=private, GetAccess=public)
+        Schema
+    end
+
+    methods
+        function obj = RecordBatchStreamReader(filename)
+            arguments
+                filename(1, 1) string {mustBeNonzeroLengthText}
+            end
+            args = struct(Filename=filename);
+            proxyName = "arrow.io.ipc.proxy.RecordBatchStreamReader";
+            obj.Proxy = arrow.internal.proxy.create(proxyName, args);
+        end
+
+        function schema = get.Schema(obj)
+            proxyID = obj.Proxy.getSchema();
+            proxyName = "arrow.tabular.proxy.Schema";
+            proxy = libmexclass.proxy.Proxy(ID=proxyID, Name=proxyName);
+            schema = arrow.tabular.Schema(proxy);
+        end
+
+        function tf = hasnext(obj)
+            tf = obj.Proxy.hasNextRecordBatch();
+        end
+
+        function tf = done(obj)
+            tf = ~obj.Proxy.hasNextRecordBatch();
+        end
```

Comment:
I'm not familiar with MATLAB but do they require APIs for iterator (or something)?

If we don't require this, it may be better that we don't provide them... If we don't have them, we can simplify our implementation.

Reply:
There is no specific APIs for iteration in MATLAB (i.e. there is no `for each` style iteration). So, we were trying to emulate this behavior using a "while has more data"-style loop. This pattern of iteration is used elsewhere in MATLAB.

Does that answer your question?

Reply:
Unfortunately, no - MATLAB does not support variable assignment like `while record_batch = reader.read()` inside of a `while` loop expression, either. If that syntax was supported, I agree it would be preferable.

Reply:
I see!